
// 套装属性
message SetEquipment{
	required int32 no = 1;  //套装NO
	optional int32 num = 2;  //套装激活数量
}

//强化数据格式
message EnhanceDataFormat{
	optional int32  before_lv = 1; //强化前等级
	optional int32 after_lv= 2; //强化后等级
	optional int32 cost_coin = 3; //强化消耗
}

// 装备
message EquipmentPB{
	required string id = 1;             // 装备ID
	optional int32 no = 2;              // 装备NO
	optional int32 strengthen_lv = 3;   // 装备强化等级
	optional int32 awakening_lv = 4;    // 装备觉醒等级
	optional int32 nobbing_effect =7;   // 装备锤炼效果
	optional int32 hero_no = 5;         // 装备武将编号
	optional SetEquipment set = 6;      // 套装
	repeated EnhanceDataFormat data= 8; // 装备强化过程
	optional bool is_guard = 9;         // 装备是否驻守
    repeated EquAttr main_attr = 10;
    repeated EquAttr minor_attr = 11;
	optional int64 prefix= 12;         // 装备前缀
    optional int32 attr_id = 13; // 如果是掉落装备，则与equipment_config中的attrid一致。 如果是商店购买的装备，则是根据幸运值计算后的结果。
	optional int64 exp = 14;         // 装备是否驻守
}

message EquAttr{
    required int32 attr_value_type = 1;
    required float attr_value = 2;
    required float attr_increment = 3;
    required int32 attr_type = 4;
}

